### ISSUE TITLE
"RemovingTag" event isn't fired.

### ISSUE THREAD

**Issue by Alpalhao** [mganss/HtmlSanitizer#58]
Hi,
After updating to HtmlSanitizer with AngleSharp, we detect the follow behaviour:

When sanitizing a text only containing a element "script" or "style", the RemovingTag event isn't fired.

Here's a test case:

```
        [Test]
        public void RemoveEventForNotAllowedTag_ScriptTag()
        {
            RemoveReason? actual = null;
            var s = new HtmlSanitizer();
            s.RemovingTag += (sender, args) =>
            {
                actual = args.Reason;
            };
            s.Sanitize("<script>alert('Hello world!')</script>");
            Assert.That(actual, Is.EqualTo(RemoveReason.NotAllowedTag));
        }

        [Test]
        public void RemoveEventForNotAllowedTag_StyleTag()
        {
            RemoveReason? actual = null;
            var s = new HtmlSanitizer();
            s.RemovingTag += (sender, args) =>
            {
                actual = args.Reason;
            };
            s.Sanitize("<style> body {background-color:lightgrey;}</style>");
            Assert.That(actual, Is.EqualTo(RemoveReason.NotAllowedTag));
        }

```

Adding another tag to the text, works fine.
```

        [Test]
        public void RemoveEventForNotAllowedTag_ScriptTagAndSpan()
        {
            RemoveReason? actual = null;
            var s = new HtmlSanitizer();
            s.RemovingTag += (sender, args) =>
            {
                actual = args.Reason;
            };
            s.Sanitize("<span>Hi</span><script>alert('Hello world!')</script>");
            Assert.That(actual, Is.EqualTo(RemoveReason.NotAllowedTag));
        }

```
 
Thanks

**Comment by mganss** [comment]
This happens because `<style>` and `<script>` in isolation are parsed into the head portion of the HTML document. But we are operating only on the body portion. As a workaround you can wrap the string to be sanitized in `<body>...</body>`, forcing the elements into the body.

This also means you cannot currently sanitize complete HTML documents, only (body) fragments. I know this is not an ideal situation, but I have no idea how to fix this right now without making the sanitizer overly complex and thus error prone. Suggestions are very welcome :smile:

**Comment by 304NotModified** [comment]
Was this also in v2? 



> As a workaround you can wrap the string to be sanitized in <body>...</body>, forcing the elements into the body.

You can do that also in this lib?

**Comment by mganss** [comment]
This was not in v2 because CsQuery is not as sophisticated a parser as AngleSharp.

I've added wrapping the input in `<body>...</body>` so the workaround will no longer be necessary in the next version (though doesn't do any harm either).

**Comment by Alpalhao** [comment]
Thanks Michael.

Any idea when this fix will be released?

**Comment by mganss** [comment]
Released as 3.1.93.
